### ISSUE TITLE
Suboptimal block sizes

### ISSUE THREAD

**Issue by lnicola** [seanmonstar/futures-fs#3]
While `8192 b` seems like a reasonable default for reading files, on modern systems it might be a bit too small, especially for sequential access. We should consider bumping it up a bit.

I ran a couple of tests on my NAS:

```text
8 kb (default)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3165M  100 3165M    0     0  87.9M      0  0:00:36  0:00:36 --:--:-- 79.8M
100 3165M  100 3165M    0     0  85.5M      0  0:00:37  0:00:37 --:--:-- 79.1M
100 3165M  100 3165M    0     0  87.9M      0  0:00:36  0:00:36 --:--:-- 79.0M

32 kb
100 3165M  100 3165M    0     0   197M      0  0:00:16  0:00:16 --:--:--  168M
100 3165M  100 3165M    0     0   166M      0  0:00:19  0:00:19 --:--:--  167M
100 3165M  100 3165M    0     0   175M      0  0:00:18  0:00:18 --:--:--  167M

128 kb
100 3165M  100 3165M    0     0   175M      0  0:00:18  0:00:18 --:--:--  175M
100 3165M  100 3165M    0     0   175M      0  0:00:18  0:00:18 --:--:--  177M
100 3165M  100 3165M    0     0   186M      0  0:00:17  0:00:17 --:--:--  181M

nginx (32 kb)
100 3165M  100 3165M    0     0   143M      0  0:00:22  0:00:22 --:--:--  140M
100 3165M  100 3165M    0     0   158M      0  0:00:20  0:00:20 --:--:--  141M
100 3165M  100 3165M    0     0   150M      0  0:00:21  0:00:21 --:--:--  142M
```

Note that I'm using ZFS, which doesn't support `sendfile()`. The `128 kb` value is the filesystem preferred buffer size (`st_blksize` in `struct stat`).

EDIT: notice how `hyper` with `futures-fs` was faster than `nginx` 👍.

**Comment by seanmonstar** [comment]
Yea, good point! I picked 8kb as its a nice default for packet size in networking. I think there's perhaps two things we could do, maybe one or the other, or both:

1. Have a configuration option for `FsPool` (or just `read`?) on preferred buffer/read size.
2. Check for `st_blksize` on `cfg(unix)` before starting to read a file.

Perhaps the best is to have the config option, and prefer that over `st_blksize` if set, so people can still explicitly say things like "even though my file system likes 128kb buffers, I don't want more than 32kb of memory used at a time". What do you think?

**Comment by lnicola** [comment]
Hmm, wasn't this closed in #5?
